src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_USER_AGE = 18;
+    public static final int MIN_LOGIN_CHARACTERS = 6;
+    public static final int MIN_PASSWORD_CHARACTERS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("User login can't be equal null");
+        } else if (user.getLogin().length() < MIN_LOGIN_CHARACTERS) {
+            throw new RegistrationException("Login length is "
+                    + user.getLogin().length() + ", minimal length " + MIN_LOGIN_CHARACTERS);
+        }
```

Comment:
```suggestion
            throw new RegistrationException("User login can't be equal null");
        } 
        if (user.getLogin().length() < MIN_LOGIN_CHARACTERS) {
            throw new RegistrationException("Login length is "
                    + user.getLogin().length() + ", minimal length " + MIN_LOGIN_CHARACTERS);
        }
```

you don't need 'else' since inside if you're throwing an exception. fix in all similar places